projects/igniteui-angular/src/lib/directives/dragdrop/dragdrop.directive.ts

```diff
@@ -794,13 +792,37 @@ export class IgxDropDirective {
     @HostBinding('class.dragOver')
     public dragover = false;
 
-    constructor(public element: ElementRef, private _renderer: Renderer2) {
+    /**
+     * @hidden
+     */
+    protected _destroy = new Subject<boolean>();
+
+    constructor(public element: ElementRef, private _renderer: Renderer2, private _zone: NgZone) {
     }
 
+    ngOnInit() {
+        this._zone.runOutsideAngular(() => {
+            fromEvent(this.element.nativeElement, 'igxDragEnter').pipe(takeUntil(this._destroy))
+                .subscribe((res) => this.onDragEnter(res as CustomEvent<IgxDragCustomEventDetails>));
```

Comment:
IMO it would be better to call the event handlers inside the Angular zone, i.e.
`.subscribe(res => this._zone.run(() => this.onDragEnter(res as CustomEvent<IgxDragCustomEventDetails>)))`

Reply:
@rkaraivanov, as @skrustev has mentioned, calling the event handlers inside the Angular zone has an unpleasant impact on the drag performance, primarily in IE and Edge (you can check the grid performance demo). I would rather leave it the way it is.